WordPress/Classes/ViewRelated/Blog/Blog Dashboard/ViewModel/BlogDashboardViewModel.swift

```diff
@@ -130,6 +143,42 @@ private extension BlogDashboardViewModel {
             scrollView.setContentOffset(position, animated: false)
         }
     }
+
+    // In case a draft is saved and the drafts card
+    // is not appearing, we show it.
+    @objc func showDraftsCardIfNeeded() {
+        guard let currentPostsInfo = currentPostsInfo else {
+            return
+        }
+        if currentPostsInfo.hasDrafts == false {
+            currentPostsInfo.hasDrafts = true
+            loadCardsFromCache()
+        }
+    }
+
+    // In case a post is scheduled and the scheduled card
+    // is not appearing, we show it.
+    @objc func showScheduledCardIfNeeded() {
+        guard let currentPostsInfo = currentPostsInfo else {
+            return
+        }
+        if currentPostsInfo.hasScheduled == false {
+            currentPostsInfo.hasScheduled = true
+            loadCardsFromCache()
+        }
+    }
+
+    // In case a post is published and create_first card
+    // is showing, we replace it with the create_next card.
+    @objc func showNextPostCardIfNeeded() {
```

Comment:
Maybe we should add methods for when posts are deleted?

Reply:
That's a good idea, but I don't think it will be as simple as that unfortunately. 
As mentioned in https://github.com/wordpress-mobile/WordPress-iOS/pull/18351#issuecomment-1098921097, I will tackle this in a different PR.

Reply:
The posts aren't instantly removed from the card because `PostService` marks it as "trashed" only after the API call finishes. This is why it stays a while displayed in the card, because in the eyes of the fetchController it is still in CoreData.

The way `PostListViewController` handles this more gracefully is by keeping in memory a list of all "recently trashed posts". Recently trashed posts are posts that have been deleted but the delete API hasn't finished yet. For these, the ViewController displays the restore post cell.

Triggering a call to `loadCardsFromCache` will not fix this, because the post will still be in CoreData and returned by the fetch request.

What I'm planning to explore is to adjust the fetch request in a way to not return "recently trashed" posts. This should fix our problem. And if there's only one post and it gets deleted, this should automatically handle displaying the next post card.